### PR TITLE
Verbsets

### PR DESCRIPTION
**Pull request by sdee**
-Allows user to choose the set of verb to use
-Displays tooltip that shows definition, if the data set has definition. Right now only the "100 most common" verbs have definitions.
-Verbs without definition have the same format as before (an array). Ones with definitions are a hash with the verb as the key and the definition as the value.

### CODE REVIEW

File: client/src/App.css [hunk 1]
```diff
@@ -111,7 +111,16 @@ section.card div {
 .questionNum {
   font: 50px 'Helvetica', sans-serif;
 }
+
 .overlay {
   color: #bfbfbf;
   padding-left: 2px;
 }
+
+.def_glyph {
```
Comment (jagatfx) [inline client/src/App.css]: @sdee dashes are preferred for css class names versus underscores. Also, .questionNum uses camel case, so this is introducing yet another convention. I recommend deciding on one and aligning all to that convention.

Comment (jagatfx) [inline client/src/App.css]: http://stackoverflow.com/questions/7560813/why-are-dashes-preferred-for-css-selectors-html-attributes

File: services/quiz.js [hunk 2]
```diff
@@ -11,6 +13,12 @@ const FILTER_ALL = 1;
 const FILTER_NONE = 2;
 const FILTER_BYCASE = 3;
 
+const verbSets = {
+	"default": verbs,
```
Comment (jagatfx) [inline services/quiz.js]: @sdee don't use quotes for property name

File: services/quiz.js [hunk 3]
```diff
@@ -109,14 +117,26 @@ function generateConjugation(verb, pronoun, moodTense) {
 	return null;
 }
 
-function generateQuestion() {
+function generateQuestion(verbSet) {
 	const moodTense = chooseMoodTense();
 	const pronoun = choosePronoun();
-	const verb = chooseVerb();
+	const data = verbSets[verbSet];
```
Comment (jagatfx) [inline services/quiz.js]: @sdee consider changing variable name to something meaningful

File: services/quiz.js [hunk 4]
```diff
@@ -109,14 +117,26 @@ function generateConjugation(verb, pronoun, moodTense) {
...
+	let verbs=[];
+	let hasDefinition = false;
+	if (data.constructor === Array){
```
Comment (jagatfx) [inline services/quiz.js]: @sdee I don't like this approach (hacky). Very non-intuitive way of organizing the verb sets. I recommend on deciding on a format in the verb files and sticking with it.

File: services/quiz.js [hunk 5]
```diff
@@ -109,14 +117,26 @@ function generateConjugation(verb, pronoun, moodTense) {
...
+		verbs = Object.keys(data);
+		hasDefinition = true;
+	}
+
+	const verb = chooseVerb(verbs);
 	const reflexive = isReflexive(verb);
 	const question = {
 		pronoun: pronoun.name,
 		verb: verb,
+		definition: hasDefinition ? data[verb] : '',
```
Comment (jagatfx) [inline services/quiz.js]: @sdee you can get rid of hasDefinition and just use data[verb] here. If it is defined use it, if not use empty string.

**Comment by jagatfx** [comment]
@sdee fyi, pushed some small lint fixes

**Comment by jagatfx** [comment]
@sdee the functionality works well. Other than my complaint about the dual system for verb data files and the way you check for it, I am +1 for merge.

**Comment by sdee** [comment]
Merging. I created a ticket in trello for devising a better verb set data format. I say we discuss right after we finish 1.0. It's easy to change the output of the scripts I'm using to parse the data.

**Pull request merged**